### DISCUSSION TITLE
destructuring in zustand

### DISCUSSION THREAD

**Discussion by AdrianEasyOze** [pmndrs/zustand#2041] [Q&A] [answered]
Hi, I am starting out using zustand and I am wondering why in the documentation the extraction of fields is always presented this way:
1.
```
const popup = usePopupStore(state => state.popup);
const setPopup = usePopupStore(state => state.setPopup);
```
instead of this way 
2.
```
  const { popup,setPopup } = usePopupStore();
```

Are these notations equivalent? Is there any difference between them in terms of how they work?

**Comment by dai-shi** [comment, marked as answer]
They are equivalent if there are only two properties in the store.

Usually, people put more properties and without the selector, it will cause extra re-renders.

We have another library [valtio](https://github.com/pmndrs/valtio) to address the issue. It can optimize re-renders with the destructuring notion.
